sentry/src/main/java/io/sentry/GsonSerializer.java

```diff
@@ -119,6 +133,13 @@ public GsonSerializer(
     return gson.fromJson(reader, Session.class);
   }
 
+  @Override
+  public Transaction deserializeTransaction(Reader reader) {
```

Comment:
can we make this generic too? `<T> T` so every `deserialize<T>(reader)` would work?

Reply:
We can although the signature won't be as nice as with `serialize` cause we need to pass class:

```
<T> T deserialize(Reader reader, Class<T> clazz);
```

Reply:
true, in Kotlin we can do `fun <reified T> deserialize(reader)` so we could `deserialize<Type>(reader)` but well...
still, I'd be in to `<T> T deserialize(Reader reader, Class<T> clazz);` so we don't need to add new methods again, wdyt?

Reply:
Agreed! Lets just do it in following PR so that this one does not inflate too much.

Reply:
can we do this on this PR or another one but still be released along with this PR? otherwise we'd need another breaking change.